biggraphite/drivers/elasticsearch.py

```diff
@@ -286,20 +285,116 @@ def delete_directory(self, name):
+        return 'wildcard', value
+
+    @staticmethod
+    def __parse_regexp_component(component):
+        """Given a complex component, this builds a constraint."""
+        regex = ""
+        for subcomponent in component:
+            if isinstance(subcomponent, bg_glob.AnySequence):
+                regex += ".*"
+            elif isinstance(subcomponent, six.string_types):
+                regex += subcomponent
+            elif isinstance(subcomponent, bg_glob.CharNotIn):
+                regex += '[^' + ''.join(subcomponent.values) + ']'
+            elif isinstance(subcomponent, bg_glob.CharIn):
+                regex += '[' + ''.join(subcomponent.values) + ']'
+            elif isinstance(subcomponent, bg_glob.SequenceIn):
+                if subcomponent.negated:
+                    regex += '.*'
+                else:
+                    regex += '(' + '|'.join(subcomponent.values) + ')'
+            elif isinstance(subcomponent, bg_glob.AnyChar):
+                regex += '.'
+            else:
+                raise GlobError("Unhandled type '%s'" % subcomponent)
+        return 'regexp', regex
+
+    @staticmethod
+    def __parse_simple_component(component):
+        """Given a component with a simple type, this builds a constraint."""
+        value = component[0]
+        if isinstance(value, bg_glob.AnySequence):
+            return None, None  # No constrain
+        elif isinstance(value, six.string_types):
+            return 'term', value
+        elif isinstance(value, bg_glob.CharNotIn):
+            return 'regexp', '[^' + ''.join(value.values) + ']'
+        elif isinstance(value, bg_glob.CharIn):
+            return 'regexp', '[' + ''.join(value.values) + ']'
+        elif isinstance(value, bg_glob.SequenceIn):
+            return 'terms', value.values
+        elif isinstance(value, bg_glob.AnyChar):
+            return 'wildcard', '?'
+        else:
+            raise GlobError("Unhandled type '%s'" % value)
+
+    @staticmethod
+    def __parse_complex_component(component):
+        def is_wildcard_sub_component(sub_component):
+            return any([isinstance(sub_component, bg_glob.AnySequence),
+                        isinstance(sub_component, bg_glob.AnyChar),
+                        isinstance(sub_component, six.string_types)
+            ])
+
+        if all(is_wildcard_sub_component(sub_c) for sub_c in component):
+            return _ElasticSearchAccessor.__parse_wildcard_component(component)
+        return _ElasticSearchAccessor.__parse_regexp_component(component)
+
+    def __glob_names(self, glob, aggregate_dirs=False):
+        # TODO really implement it
+        if aggregate_dirs == True:
+            return []
+
+        if glob == "":
+            return []
+
+        search = Search().using(self.client).index(self._index).source('name')
+        # TODO Only if no globstars !!!
+        # en attendant, we assert no globstars
```

Comment:
for globstart we can probably use a simple regexp on name (which means special handling for AnySequence/AnyChar)

Reply:
Ok. Still need to be implemented.